cmd/gateway-azure.go

```diff
@@ -79,6 +79,7 @@ func s3MetaToAzureProperties(s3Metadata map[string]string) (storage.BlobMetadata
 			// Strip header prefix, to let Azure SDK
 			// handle it for storage.
 			k = strings.Replace(k, "X-Amz-Meta-", "", 1)
+			k = strings.Replace(strings.ToLower(k), "-", "_", -1)
```

Comment:
We decided not to convert all headers into C# friendly names.   This PR requires further discussion.

Reply:
Well our assumptions are wrong. I don't see why we decided what we decided. There are clearly many applications that breaks without this change.

Reply:
This is known behavior. That is consciously decided.  However I blocked it for further discussion.

Reply:
Can you clarify here why such an assumption is made? I am not aware why we decided to do this.

Reply:
I would like to know the assumptions and update relevant documents on why this was assumed you to be okay and expected behavior.